src/hooks/useAlternateLinks.ts

```diff
@@ -0,0 +1,41 @@
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { getOrigin } from '../helpers/getOrigin';
+import { getSourceUrl } from '../helpers/getSourceUrl';
+import { useRewrites } from '../hooks/useRewrites';
+
+interface AlterNateLinks {
+  href: string;
+  hrefLang: string;
+}
+
+export const useAlternateLinks = (currentLocale: string): AlterNateLinks[] | null => {
+  const { basePath, pathname, locales } = useRouter();
+  const rewrites = useRewrites();
+  const [alternateLinks, setAlternateLinks] = useState<AlterNateLinks[]>([]);
+  const path = `${basePath ?? ''}${pathname}`;
+
+  useEffect(() => {
+    const origin = getOrigin();
+
+    /*
+      If we are at the root '/' of the website we should build alternate
+      links for all the locales
+    */
+    const alternateLocales =
+      pathname === '/' ? locales : locales.filter((lang) => lang !== currentLocale);
+    const alternateLinks = alternateLocales.map((lang) => {
+      const alternateLink = getSourceUrl({ rewrites, locale: lang, path });
+      return { href: `${origin}${alternateLink}`, hrefLang: lang };
+    });
+    /*
+      If we are at the root '/' of the website we should add an alternate
+      link with hreflang = 'x-default'
+    */
+    setAlternateLinks(
+      pathname === '/' ? [...alternateLinks, { href: path, hrefLang: 'x-default' }] : alternateLinks
+    );
```

Comment:
I'd skip this for now, and later add an option or something for handling this. And if that option is enabled, then the `x-default` value should be included on _all_ pages, pointing to the specified page that acts as a language selector.

Reply:
@eemeli Helio and I just had a chat and we are planning to try to remove the "localeless" URLs (an implementation idea here: https://github.com/vercel/next.js/discussions/18419#discussioncomment-327128) - this means that `x-default` would only be required on the homepages (/ and localized version) unless you are seeing something we are missing

Reply:
👍 If an API is provided that generates the normal hreflang links for insertion into e.g. a `next/head` component, then it might be easiest for a user to add the `x-default` links explicitly to the relevant page.

Reply:
@eemeli  just to make sure I understand, you are suggesting a better way to handle `x-default` for the "default Next.js" behavior? If so, my only concern with the "localeless" URLs is that they are really not `x-default` (at least the Google definition). If I understood correctly, `x-default` should be able to change locale dynamically which is not the case for the way those pages behave in Next.js - it's actually just the default locale.

Reply:
To rephrase, I meant that if the API provided by this package is composable (i.e. does not prevent a user from adding their own things to the `<head>`), then it should be fine for an MVP solution to _not_ solve what the user should do with `x-default` hreflang links, and to leave that to be handled by the user on a page-by-page basis.